src/lib/operators/sort.cpp

```diff
@@ -33,19 +20,78 @@ const std::string& Sort::name() const {
 std::shared_ptr<AbstractOperator> Sort::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  return std::make_shared<Sort>(copied_input_left, _column_id, _order_by_mode, _output_chunk_size);
+  return std::make_shared<Sort>(copied_input_left, _sort_definitions, _output_chunk_size);
 }
 
 void Sort::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> Sort::_on_execute() {
-  _impl = make_unique_by_data_type<AbstractReadOnlyOperatorImpl, SortImpl>(
-      input_table_left()->column_data_type(_column_id), input_table_left(), _column_id, _order_by_mode,
-      _output_chunk_size);
-  return _impl->_on_execute();
+  _sort_definition_data_types.resize(_sort_definitions.size());
+  for (auto column_id = ColumnID{0}; column_id < _sort_definitions.size(); ++column_id) {
+    auto sort_definition = _sort_definitions[column_id];
+    _sort_definition_data_types[column_id] = input_table_left()->column_data_type(sort_definition.column);
+  }
```

Comment:
You might use `std::transform` here to make the code more readable.

Reply:
Good idea, thank you! Done in 40bc0bf1d6c66f646268bb8d1f601cf33e21bd44